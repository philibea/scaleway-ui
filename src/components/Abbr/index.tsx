import styled from '@emotion/styled'
import React from 'react'

// See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/abbr#default_styling
// for the default styling of the <abbr> element.
const StyledAbbr = styled.abbr`
  font-variant: none;
`

export type AbbrProps = {
  children: string
  title?: string
}

const Abbr = ({ children, title }: AbbrProps): JSX.Element => (
  <StyledAbbr title={title}>{children}</StyledAbbr>
)

export default Abbr
