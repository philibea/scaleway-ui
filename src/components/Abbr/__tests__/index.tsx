import React from 'react'
import Abbr from '..'
import { shouldMatchEmotionSnapshot } from '../../../helpers/jestHelpers'

describe('Abbr', () => {
  test('renders correctly', async () =>
    shouldMatchEmotionSnapshot(<Abbr>Abbreviation</Abbr>))

  test('renders with title', async () =>
    shouldMatchEmotionSnapshot(<Abbr title="Abbreviation">Abbr</Abbr>))
})
