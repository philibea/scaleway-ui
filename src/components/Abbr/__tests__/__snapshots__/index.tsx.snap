// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abbr renders correctly 1`] = `
<DocumentFragment>
  .cache-5zdete {
  font-variant: none;
}

<abbr
    class="cache-5zdete elquw7y0"
  >
    Abbreviation
  </abbr>
</DocumentFragment>
`;

exports[`Abbr renders with title 1`] = `
<DocumentFragment>
  .cache-5zdete {
  font-variant: none;
}

<abbr
    class="cache-5zdete elquw7y0"
    title="Abbreviation"
  >
    Abbr
  </abbr>
</DocumentFragment>
`;
